packages/toolkit/src/view/knowledge/components/FileChunks.tsx

```diff
@@ -0,0 +1,91 @@
+import React from "react";
+
+import { Icons, Nullable } from "@instill-ai/design-system";
+
+import {
+  useGetFileContent,
+  useListChunks,
+} from "../../../lib/react-query-service/knowledge";
+import {
+  Chunk,
+  KnowledgeBase,
+  KnowledgeFile,
+} from "../../../lib/react-query-service/knowledge/types";
+import ChunkCard from "./ChunkCard";
+
+type FileChunksProps = {
+  file: KnowledgeFile;
+  knowledgeBase: KnowledgeBase;
+  accessToken: Nullable<string>;
+  expanded: boolean;
+  onToggleExpand: (fileUid: string) => void;
+  onChunkClick: (file: KnowledgeFile, chunk: Chunk) => void;
+  onRetrievableToggle: (
+    chunkUid: string,
+    currentValue: boolean,
+  ) => Promise<void>;
+};
+
+const FileChunks: React.FC<FileChunksProps> = ({
+  file,
+  knowledgeBase,
+  accessToken,
+  expanded,
+  onToggleExpand,
+  onChunkClick,
+  onRetrievableToggle,
+}) => {
+  const { data: chunks, isLoading: isLoadingChunks } = useListChunks({
+    kbId: knowledgeBase.kbId,
+    accessToken: accessToken || null,
+    enabled: expanded,
+    ownerId: knowledgeBase.ownerName,
+    fileUid: file.fileUid,
+  });
+
+  const { data: fileContent } = useGetFileContent({
+    fileUid: file.fileUid,
+    kbId: knowledgeBase.kbId,
+    accessToken: accessToken || null,
+    enabled: expanded,
+    ownerId: knowledgeBase.ownerName,
+  });
```

Comment:
@thewbuk Please don't deconstruct the react query return. It will reduce readability